pinot-common/src/main/java/org/apache/pinot/common/request/context/FunctionContext.java

```diff
@@ -35,12 +35,14 @@ public enum Type {
   private final Type _type;
   private String _functionName;
   private final List<ExpressionContext> _arguments;
+  private final boolean _isDistinct;
 
-  public FunctionContext(Type type, String functionName, List<ExpressionContext> arguments) {
+  public FunctionContext(Type type, String functionName, List<ExpressionContext> arguments, boolean isDistinct) {
```

Comment:
This is single-stage engine context. Do we need to push this to the single stage side? I think we will need to collect all records for a partition, sorting the records, then apply the distinct and aggregation?

Reply:
this is also used in multi-stage engine to init the function.